app/WebParsing/UtsgJsonParser.hs

```diff
@@ -0,0 +1,157 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+module WebParsing.UtsgJsonParser
+     (getJSON) where
+-}
+
+import Data.Aeson
+import Data.List
+import Database.Tables
+import WebParsing.PrerequisiteParsing
+import Database.Persist.Sqlite (runSqlite, insertMany_)
+import qualified Data.HashMap.Lazy as M
+import qualified Data.Text as T
+import Config (databasePath)
+import Network.HTTP.Conduit (simpleHttp)
+
+-- | Converts 24-hour time into a double
+-- | Assumes times are rounded to the nearest hour
+getHourVal :: String -> Double
+getHourVal time = (read $ take 2 time :: Double)
+
+-- | Converts a weekday into a double
+-- | Monday to Friday becomes 0.0 to 4.0
+getDayVal :: String -> Double
+getDayVal day = case day of
+                    "MO" -> 0.0
+                    "TU" -> 1.0
+                    "WE" -> 2.0
+                    "TH" -> 3.0
+                    "FR" -> 4.0
+
+-- | Takes a day and start/end times then generates a set of 30-minute timeslots
+getTimeSlots :: String -> String -> String -> [Time]
+getTimeSlots day start end = do
+    let dayDbl = getDayVal day
+        startDbl = getHourVal start
+        endDbl = getHourVal end
+    [Time [dayDbl, timeDbl] | timeDbl <- [startDbl, (startDbl + 0.5) .. (endDbl - 0.5)]]
+
+getNestedTimeSlots :: [Maybe T.Text] -> [Maybe T.Text] -> [Maybe T.Text] -> [[Time]]
+getNestedTimeSlots [Nothing] [Nothing] [Nothing] = [[]]
```

Comment:
It seems this function assumes the input lists are just a single `Nothing`, or a list of `Just`s. If that's the case, then the proper input type isn't `[Maybe T.Text]`, but instead `Maybe [T.Text]`.

On the other hand, perhaps the lists could contain a mixture of Justs and Nothings, which might mean there shouldn't be two pattern-matching rules, but instead a more robust function passed to `zipWith3`.